doc/whats_new/v1.3.rst

```diff
@@ -162,6 +162,13 @@ Changelog
 :mod:`sklearn.metrics`
 ......................
 
+- |Efficiency| The computation of the expected mutual information in
+  :func:`metrics.adjusted_mutual_info_score` is now faster when the number
+  of labels is large and its memory usage is reduced in general.
+  :pr:`25713` by :user:`Kshitij Mathur <Kshitij68>`,
+  :user:`Guillaume Lemaitre <glemaitre>` and
+  :user:`Jérémie du Boisberranger <jeremiedbb>`.
```

Comment:
Please also credit @OmarManzoor who started the implementation that introduced the use of memory views in the inner loops.